File: triangolo.cpp
#include "triangolo.h"

double triangolo::numeroFisso = 0.289 ;

double triangolo::getfisso() const{
    return numeroFisso;
}

/* CALCOLO DISTANZA TRA I PUNTI TRA 1-2 1-3 2-3 */
double triangolo::perimetro() const{
    double per = 0 ;
    per = per + punto::distanceTwoPoints(*pt[0],*pt[1]) + punto::distanceTwoPoints(*pt[0],*pt[2]) + punto::distanceTwoPoints(*pt[1],*pt[2]) ;

    return per;
}

double triangolo::lato() const {
    if( punto::distanceTwoPoints(*pt[0],*pt[1]) == punto::distanceTwoPoints(*pt[0],*pt[2]) == punto::distanceTwoPoints(*pt[1],*pt[2])){
        return punto::distanceTwoPoints(*pt[0],*pt[1]);
    }
    else return 0;
}

double triangolo::area( ) const {
    double base = punto::distanceTwoPoints(*pt[0],*pt[1]) ;

    double h = retta::distancePuntoRetta(*pt[2],retta::rettaFromTwoPoints(*pt[0],*pt[1]));

    return (base*h)/2;
}

File: java/triangolo.java
import java.util.Vector;

public class triangolo extends poligono{
	private static double numeroFisso;

	public triangolo(int i , Vector<punto> p){
		super(i,p);
	}

	public triangolo() {
		super(0);	
	}

	public double getfisso(){
		return numeroFisso;
	}

	public double perimetro(){
		return (pt.get(0)).distancefromtwopoints(pt.get(1)) + (pt.get(1)).distancefromtwopoints(pt.get(2)) + (pt.get(0)).distancefromtwopoints(pt.get(2));
	}

	public double area(){
		double base = (pt.get(0)).distancefromtwopoints(pt.get(1)) ;
		double h = ((pt.get(0)).rettafromtwopoints(pt.get(1))).distancePuntoRetta(pt.get(2));
		return (base*h)/2;
	}

	public double lato(){
		double a = ((pt.get(1)).distancefromtwopoints(pt.get(2)));
		double b = ((pt.get(0)).distancefromtwopoints(pt.get(2)));
		double c = ((pt.get(0)).distancefromtwopoints(pt.get(1)));

	    if(a == b ){
	    	if(a == c)
	        	return (pt.get(1)).distancefromtwopoints(pt.get(2));
	    }
	    return 0; //non regolare
	}


}